pkgs/applications/version-management/jiri/default.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv
+, lib
+, fetchurl
+, fetchzip
+, unzip
+, autoPatchelfHook
+, version ? "unstable-2021-06-15"
+, sources ?
+  let
+    base = "https://chrome-infra-packages.appspot.com/dl/fuchsia/tools/jiri";
+    x86_64 = "amd64";
+    revision = "fcdfde57a8309823f6899264dd9c009a2e6fb01e";
+    version = "unstable-2021-06-15";
+  in
+  {
+    "${version}-x86_64-linux" = fetchurl {
+      name= "git_revision-${revision}.zip";
+      url = "${base}/linux-${x86_64}/+/git_revision:${revision}";
+      sha256 = "11f35gni1dzb5lvdy1dcx9zk0m5jb2bb9jmq2c8kx2brrhcdlzdv";
+    };
+    "${version}-x86_64-darwin" = fetchurl {
+      name= "git_revision-${revision}.zip";
+      url = "${base}/mac-${x86_64}/+/git_revision:${revision}";
+      sha256 = "c9dd2d345e91418a904094b4f0d64a0e872635c1aef6ed030c345632cf1cc626";
+    };
+  }
+}:
+
+assert version != null && version != "";
+assert sources != null && (builtins.isAttrs sources);
+
+stdenv.mkDerivation {
+  pname = "jiri";
+  inherit version;
+
+  src = sources."${version}-${stdenv.hostPlatform.system}" or (throw "unsupported version/system: ${version}/${stdenv.hostPlatform.system}");
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    unzip
+  ];
+
+  installPhase = ''
+    mkdir -p $out
+    cp -R * $out/
+    chmod +x $out/jiri
+  '';
+
+  unpackPhase = ''
+    mkdir -p $out
+    unzip -d $out $src
+  '';
```

Comment:
This should be done automatically.

Reply:
When I build it with the unpack phase commented out (i.e. letting things happen automatically), I get an error that apparently nothing is happening.

```
$ nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'                                                                                                                                      
these derivations will be built:
  /nix/store/i065l75n4nns339r23vbl8vjqw2xalfb-jiri-unstable-2021-06-15.drv
building '/nix/store/i065l75n4nns339r23vbl8vjqw2xalfb-jiri-unstable-2021-06-15.drv'...
unpacking sources
unpacking source archive /nix/store/blyvls9fd4ipjsp98p92qfixarqbwgvn-git_revision-fcdfde57a8309823f6899264dd9c009a2e6fb01e.zip
unpacker appears to have produced no directories
builder for '/nix/store/i065l75n4nns339r23vbl8vjqw2xalfb-jiri-unstable-2021-06-15.drv' failed with exit code 1
error: build of '/nix/store/i065l75n4nns339r23vbl8vjqw2xalfb-jiri-unstable-2021-06-15.drv' failed
```

Is there something I'm missing?

Reply:
unpackPhase by default unpacks the files into a temporary build directory from where you need to copy them to $out. Maybe you need to add stripRoot = false.

Reply:
Resolved this with 

```
setSourceRoot = "sourceRoot=`pwd`";
```

since there is no subdirectories that are done in the unpack session